conan/tools/cmake/cmakedeps/templates/config.py

```diff
@@ -45,22 +45,29 @@ def template(self):
             message(FATAL_ERROR "The 'CMakeDeps' generator only works with CMake >= 3.15")
         endif()
 
+        include(${CMAKE_CURRENT_LIST_DIR}/cmakedeps_macros.cmake)
+        include(${CMAKE_CURRENT_LIST_DIR}/{{ targets_include_file }})
+        include(CMakeFindDependencyMacro)
+
         {% if is_module %}
         include(FindPackageHandleStandardArgs)
-        set({{ pkg_name }}_FOUND 1)
-        set({{ pkg_name }}_VERSION "{{ version }}")
+        set({{ file_name }}_FOUND 1)
+        set({{ file_name }}_VERSION "{{ version }}")
 
-        find_package_handle_standard_args({{ pkg_name }}
-                                          REQUIRED_VARS {{ pkg_name }}_VERSION
-                                          VERSION_VAR {{ pkg_name }}_VERSION)
-        mark_as_advanced({{ pkg_name }}_FOUND {{ pkg_name }}_VERSION)
-        {% endif %}
+        find_package_handle_standard_args({{ file_name }}
+                                          REQUIRED_VARS {{ file_name }}_VERSION
+                                          VERSION_VAR {{ file_name }}_VERSION)
+        mark_as_advanced({{ file_name }}_FOUND {{ file_name }}_VERSION)
 
-        include(${CMAKE_CURRENT_LIST_DIR}/cmakedeps_macros.cmake)
-        include(${CMAKE_CURRENT_LIST_DIR}/{{ targets_include_file }})
-        include(CMakeFindDependencyMacro)
+        foreach(_DEPENDENCY {{ '${' + file_name + '_FIND_DEPENDENCY_NAMES' + '}' }} )
+            find_dependency({{ '${_DEPENDENCY}' }} REQUIRED MODULE)
+        endforeach()
+
+        {% endif %}
 
-        foreach(_DEPENDENCY {{ '${' + pkg_name + '_FIND_DEPENDENCY_NAMES' + '}' }} )
+        foreach(_DEPENDENCY {{ '${' + file_name + '_FIND_DEPENDENCY_NAMES' + '}' }} )
```

Comment:
A bit difficult to understand, why this is not in the ``else is_module``, without the ``if _FOUND``? If ``is_module``, then does it make sense to try to find the ``NO_MODULE``? Probably I don't understand how it works, but just in case.

Reply:
This was thinking in the cmake_find_mode="both", but probably it makes no sense, I'll have a second thought to this